src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -41,7 +41,8 @@ void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo)
 
 CHIP_ERROR DefaultCheckInDelegate::GenerateRefreshKey(RefreshKeySender::RefreshKeyBuffer & newKey)
 {
-    return Crypto::DRBG_get_bytes(newKey.Bytes(), newKey.Capacity());
+    ReturnErrorOnFailure(Crypto::DRBG_get_bytes(newKey.Bytes(), newKey.Capacity()));
+    return newKey.SetLength(Crypto::kAES_CCM128_Key_Length);
```

Comment:
Why are we putting `newKey.Capacity()` bytes in there but then claiming the length is `Crypto::kAES_CCM128_Key_Length`?

Should be consistent.

Reply:
done, thanks

Reply:
done, thanks